app/data/AuditDataStoreProvider.scala

```diff
@@ -21,18 +24,25 @@ class AuditDataStore(client: AmazonDynamoDBClient, auditDynamoTableName: String)
     db.query("atomId", id).map(Audit.fromItem).toList
   }
 
-  def auditPublish(atomId: String, username: Option[String]) = {
-    val audit = Audit(atomId, "publish", None, DateTime.now().getMillis, username.getOrElse("unknown user"))
+  private def getUsername (user: PandaUser): String = {
```

Comment:
temporary hack until https://github.com/guardian/panda-hmac/pull/2 is released, then we can pattern match like:

```scala
val username = request.user match {
  case pandaUser: PandaUser => user.email
  case hmacUser: HMACAuthenticatedService => user.name
}
```